src/java/htsjdk/samtools/SRAIndex.java

```diff
@@ -0,0 +1,215 @@
+
+    /**
+     * Reports the maximum number of bins that can appear in a BAM file.
+     */
+    private static final int MAX_BINS = 37450;   // =(8^6-1)/7+1
+    /**
+     * First bin number in last level
+     */
+    private static final int SRA_BIN_INDEX_OFFSET = LEVEL_STARTS[LEVEL_STARTS.length - 1];
+
+    private static int SRA_BIN_SIZE = 16 * 1024;
+    private static int SRA_CHUNK_SIZE = 50000;
+
+    private SAMFileHeader header;
+    private SRAIterator.RecordRangleInfo recordRangleInfo;
+
+    public SRAIndex(SAMFileHeader header, SRAIterator.RecordRangleInfo recordRangleInfo) {
+        this.header = header;
+        this.recordRangleInfo = recordRangleInfo;
+    }
+
+    @Override
+    public int getLevelSize(int levelNumber) {
+        if(levelNumber == getNumIndexLevels()-1)
+            return MAX_BINS-LEVEL_STARTS[levelNumber]-1;
+        else
+            return LEVEL_STARTS[levelNumber+1]-LEVEL_STARTS[levelNumber];
+    }
+
+    @Override
+    public int getLevelForBin(Bin bin) {
+        return LEVEL_STARTS.length - 1;
+    }
+
+    @Override
+    public int getFirstLocusInBin(Bin bin) {
+        int refIndex = bin.getReferenceSequence();
+        long refOffset = recordRangleInfo.getReferenceOffsets().get(refIndex);
+        List<Chunk> chunkList = getBinChunks(bin);
+
+        if (chunkList.isEmpty()) {
+            throw new RuntimeException("Cannot find any locus in bin: bin is empty");
+        }
+
+        long minPos = Long.MAX_VALUE;
+        for (Chunk chunk : chunkList) {
+            long refChunkStart = chunk.getChunkStart() - refOffset;
+            assert(refChunkStart >= 0);
+            if (refChunkStart < minPos) {
+                minPos = refChunkStart;
+            }
+        }
+
+        return (int)minPos;
+    }
+
+    @Override
+    public int getLastLocusInBin(Bin bin) {
+        int refIndex = bin.getReferenceSequence();
+        long refOffset = recordRangleInfo.getReferenceOffsets().get(refIndex);
+        List<Chunk> chunkList = getBinChunks(bin);
+
+        if (chunkList.isEmpty()) {
+            throw new RuntimeException("Cannot find any locus in bin: bin is empty");
+        }
+
+        long maxPos = 0;
+        for (Chunk chunk : chunkList) {
+            long refChunkEnd = chunk.getChunkEnd() - refOffset - 1;
+            assert(refChunkEnd > 0);
+            if (refChunkEnd > maxPos) {
+                maxPos = refChunkEnd;
+            }
+        }
+
+        return (int)maxPos;
+    }
+
+    /**
+     * @param referenceIndex sequence of desired SAMRecords
+     * @param startPos 1-based start of the desired interval, inclusive
+     * @param endPos 1-based end of the desired interval, inclusive
+     * @return a list of bins that contain relevant data
+     */
+    @Override
+    public BinList getBinsOverlapping(int referenceIndex, int startPos, int endPos) {
+        long refOffset = recordRangleInfo.getReferenceOffsets().get(referenceIndex);
```

Comment:
`refOffset` is unused in this method -- can you explain how you are able to get the `Bin`s overlapping the requested interval without taking `refOffset` into account?